Add detailed build instructions for OpenJDK 13

SueChaplain opened the pull request:
Add missing build instructions and docker/yum artifacts.
Remove unsupported JDK yum files

[ci-skip]

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

SueChaplain commented:
@pshipton - these are untested, copied from OpenJDK12. Are you aware of any differences that are needed.

pshipton commented on buildenv/docker/jdk13/ppc64le/ubuntu16/Dockerfile:
```diff
@@ -0,0 +1,99 @@
+
+# Download and install boot JDK from AdoptOpenJDK
+RUN cd /root \
+  && wget -O bootjdk11.tar.gz "https://api.adoptopenjdk.net/v2/binary/releases/openjdk11?openjdk_impl=openj9&os=linux&arch=ppc64le&release=latest&type=jdk" \
```
jdk12 or jdk13 is needed to build jdk13. Same comment for all dockerfile's and the build instructions.

SueChaplain replied:
Fixed.

pshipton commented:
For AIX, xlc 16 is required. Plus it doesn't work yet.

pshipton commented:
@jdekonin @AdamBrousseau in case you know of other differences for builds jdk13 vs jdk12.

AdamBrousseau commented:
I don't see anything obvious based on our [variable file](https://github.com/eclipse/openj9/blob/master/buildenv/jenkins/variables/defaults.yml) other than xlc16 and bootjdks which you already pointed out. I am wondering if we should be adding these or revisiting #5276 ?

SueChaplain commented:
@pshipton 
>For AIX, xlc 16 is required. Plus it doesn't work yet, #5074

Updated to xlc 16. The section has an "under construction" notice, so I don't think we need to call out the issues with the build here.

@AdamBrousseau 
>I am wondering if we should be adding these or revisiting #5276 ?

I do remember that getting raised, but it looks like it never got resolved. I've gone ahead and updated the ones we were pointing to, but we can look to make a bigger change if everyone can agree on what is best to do.

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,654 @@
+rm -f freemarker.tgz
+```
+
+4. Download and setup the boot JDK using the latest AdoptOpenJDK v11 build.
````
These instructions still refer to jdk11, but you need jdk12 or jdk13 to build jdk13.

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
```diff
@@ -0,0 +1,654 @@
+
+A number of RPM packages are also required. The easiest method for installing these packages is to use `yum`, because `yum` takes care of any additional dependent packages for you.
+
+Download the following file: [yum_install_aix-ppc64.txt](aix/jdk11/yum_install_aix-ppc64.txt)
```
should be jdk13

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
```diff
@@ -0,0 +1,654 @@
+- [xlc/C++ 16](https://www.ibm.com/developerworks/downloads/r/xlcplusaix/)
+- x11.adt.ext
+
+You must also install the boot JDK: [Java11_AIX_PPC64](https://adoptopenjdk.net/releases.html?variant=openjdk11&jvmVariant=openj9#ppc64_aix).
```
should be jdk12 or 13

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,654 @@
+
+- Run the following commands:
+```
+wget https://go.microsoft.com/fwlink/?LinkId=532495 -O vs2013.exe
````
should be 2017 for jdk11 and above. I expect this means the link also needs updating

SueChaplain replied:
I can't find what that link should be.

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,654 @@
+wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tgz
+```
+- Before installing Visual Studio, change the permissions on the installation file by running `chmod u+x vs2013.exe`.
+- Install Visual Studio by running the file `vs2013.exe` (There is no special step required for downloading/installing VS2017. Please follow the guide of the downloaded installer to install all required components, especially for VC compiler).
````
2017

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
```diff
@@ -0,0 +1,654 @@
+- Install Visual Studio by running the file `vs2013.exe` (There is no special step required for downloading/installing VS2017. Please follow the guide of the downloaded installer to install all required components, especially for VC compiler).
+
+Not all of the shared libraries that are included with Visual Studio are registered during installation.
+In particular, the `msdia120.dll`(VS2013) or `msdia140.dll`(VS2017) libraries must be registered manually.
```
just 2017

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
```diff
@@ -0,0 +1,654 @@
+In particular, the `msdia120.dll`(VS2013) or `msdia140.dll`(VS2017) libraries must be registered manually.
+To do so, execute the following from a command prompt:
+
+**VS2013**
```
obsolete

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,654 @@
+When you have all the source files that you need, run the configure script, which detects how to build in the current build environment.
+```
+bash configure --disable-warnings-as-errors \
+               --with-toolchain-version=2013 or 2017 \
````
just 2017

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,654 @@
+               --with-toolchain-version=2013 or 2017 \
+               --with-freemarker-jar=/cygdrive/c/temp/freemarker.jar
+```
+Note: there is no need to specify --with-toolchain-version for 2017 as it will be located automatically
````
obsolete

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
```diff
@@ -0,0 +1,654 @@
+You must install a number of software dependencies to create a suitable build environment on your system:
+
+- [Xcode 9.4]( https://developer.apple.com/download/more/) (requires an Apple account to log in).
+- [macOS OpenJDK 11](https://adoptopenjdk.net/archive.html?variant=openjdk11&jvmVariant=hotspot), which is used as the boot JDK.
```
jdk12 or 13 is needed

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,654 @@
+
+```
+bash configure --with-freemarker-jar=/<my_home_dir>/freemarker.jar \
+               --with-boot-jdk=<path_to_macOS_JDK11> \
````
jdk12 or 13

pshipton replied:
Seems strange that some platforms say to use `--with-boot-jdk=` and some don't. Unless you have the boot-jdk on your path, it always needs to be specified.

SueChaplain commented:
@pshipton - think I got everything you mention (and more), except for the VS2017 wget link. Tried a few times to establish what it should be. Perhaps @vsebe can help?

pshipton commented:
Start from https://visualstudio.microsoft.com/vs/older-downloads/
You need a Microsoft account and to join the free "Visual Studio Dev Essentials" program to get the download. There isn't any direct link.

vsebe commented:
@SueChaplain - could you please also try https://aka.ms/vs/15/release/vs_community.exe (that's what the [playbook](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2017/tasks/main.yml) uses to install VS 2017)

SueChaplain commented:
@vsebe - that is indeed VS2017 by direct link: https://aka.ms/vs/15/release/vs_community.exe 

Can I just add a `wget` infront of this for the instructions?

pshipton commented:
Windows doesn't have wget by default, we should just stick with the link.

SueChaplain commented:
>Windows doesn't have wget by default, we should just stick with the link.

The build instructions walk through using Cygwin, hence we have already documented `wget` steps for compilers and freemarker. I was hoping to slot this link in place of the old one to minimise the update needed. i.e. update the following by replacing the first link:

```
wget https://go.microsoft.com/fwlink/?LinkId=532495 -O vs2013.exe
wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tgz
```

vsebe commented:
>Can I just add a wget in front of this for the instructions?

@SueChaplain, yes, you can (run in a cygwin terminal).

SueChaplain commented:
@pshipton - I think we are done on this now. Perhaps you can do one final sweep please. Thanks

pshipton commented on buildenv/aix/jdk11/yum_install_aix-ppc64.txt:
```diff
@@ -18,33 +18,33 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 
-# This file installs package dependencies for building OpenJDK V10 with OpenJ9 on AIX 64-bit systems
+# This file installs package dependencies for building OpenJDK V11 with OpenJ9 on AIX 64-bit systems
```
Not sure how this file got past the copyright check, the copyright end date should be 2019.

SueChaplain replied:
Done

pshipton replied:
The copyright end date should be 2019.

SueChaplain replied:
done

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,651 @@
+
+- Run the following commands:
+```
+wget https://aka.ms/vs/15/release/vs_community.exe 
````
The `-O vs2017.exe` is missing, so this doesn't match the instructions below. I also noticed there is another link to download 2017 used above i.e. https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&rel=15

SueChaplain replied:
Fixed and modified the earlier link to be the same

pshipton replied:
The `-O vs2017.exe` is still missing.

SueChaplain replied:
I misunderstood. Thought you wanted the instructions below it updating.

SueChaplain replied:
Done

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
```diff
@@ -0,0 +1,651 @@
+You must install a number of software dependencies to create a suitable build environment on your system:
+
+- [Xcode 9.4]( https://developer.apple.com/download/more/) (requires an Apple account to log in).
+- [macOS OpenJDK 12](https://adoptopenjdk.net/archive.html?variant=openjdk12&jvmVariant=hotspot), which is used as the boot JDK.
```
OpenJ9 is preferred over Hotspot.

SueChaplain replied:
Yep ... must be from the days when we didn't have one to use. Fixed.

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,651 @@
+To do so, execute the following from a command prompt:
+
+```
+**VS2017**
````
I don't think we need this header any more since there is only one version.

SueChaplain replied:
Removed.

pshipton commented on doc/build-instructions/Build_Instructions_V13.md:
````diff
@@ -0,0 +1,651 @@
+In particular, the `msdia140.dll`(VS2017) library must be registered manually.
+To do so, execute the following from a command prompt:
+
+```
````
These back-quotes should be removed, they mess up everything below.

SueChaplain replied:
Yep, done

SueChaplain commented on buildenv/aix/jdk13/yum_install_aix-ppc64.txt:
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2017, 2017 IBM Corp. and others
+# Copyright (c) 2019, 2019 IBM Corp. and others
```
Done

SueChaplain commented:
Final changes complete. Hope this is ready for merging @pshipton ?
